host-bin/enter-chroot

```diff
@@ -436,9 +445,21 @@ elif [ ! -f "$shares" ]; then
 downloads ~/Downloads
 EOF
 fi
+
+activeuser=$(getactiveuser)
+if [ -n "$firstrun" ]; then
+    # Store the active user for possible use in other scripts, etc.
+    activeuserfile="$(fixabslinks '/etc/crouton/user')"
+    echo $activeuser > $activeuserfile
+fi
+
 # Bind-mount the stuff in $CHROOT/etc/crouton/shares, unless NOLOGIN is set
 if [ -z "$NOLOGIN" -a -f "$shares" ]; then
-    localdownloads='/home/chronos/user/Downloads'
+    if [ "$activeuser" != "Default" ]; then
+        localdownloads=$(jq -r ".profile.info_cache | to_entries[] | select(.value.user_name == \"$activeuser\") | \"/home/chronos/\(.key)/Downloads\"" /home/chronos/Local\ State)
```

Comment:
Although 'activeuser' is populated and valid, both 'localdownloads' locations appear empty:
```
chronos@localhost ~ $ localdownloads=$(jq -r ".profile.info_cache | to_entries[] | select(.value.user_name == \"$activeuser\") | \"/home/chronos/\(.key)/Downloads\"" /home/chronos/Local\ State)

ls -l $localdownloads
/home/chronos/Default/Downloads:
total 0

/home/chronos/LockScreenAppsProfile/Downloads:
total 0
```
Yet when I list `/home/chronos/user/Downloads` it contains all of my files.

I'm not sure this routine will get you the expected results but I may not understand the full intent.

Reply:
Ah, I'm guessing you haven't logged in to multiple accounts on that Chromebook?

Even so, I expected that the `jq -r` part would not have run (`if [ "$activeuser" != "Default" ]`) and would have used `/home/chronos/user/Downloads` instead.

Would you mind checking the output of `jq -r .LastActiveUser /home/chronos/Local\ State` ? That's what `activeuser` should have been set to.

Reply:
BTW, here's what I'm getting (except for the actual email address and u- id):
```
$ activeuser=$(jq -r .LastActiveUser /home/chronos/Local\ State)
$ echo $activeuser
myemail@example.com

$ localdownloads=$(jq -r ".profile.info_cache | to_entries[] | select(.value.user_name == \"$activeuser\") | \"/home/chronos/\(.key)/Downloads\"" /home/chronos/Local\ State)
$ echo $localdownloads
/home/chronos/u-xxxxxxxe64bcd6851c24f81f6566ef0519e781cba/Downloads

$ ls $localdownloads
...files of myemail@example.com here...
```

Reply:
You're correct, I only have one account setup on this device.
Here's the output you requested with my email masked:
```
chronos@localhost ~ $ jq -r .LastActiveUser /home/chronos/Local\ State
xxxxxxxx@gmail.com
```
It could be that when I add another user account I'd get the correct $localdownloads location populated but I'm thinking it might be best not to do that yet so we can test this on devices with only one account. When I run the second part beginning with `localdownloads=...` I still get the two empty directories.

Let me know if I'm doing something wrong or if you'd like me to test something else.

Reply:
I ran the modified `enter-chroot` adding `set -x` at line [#455](https://github.com/ibrado/crouton/blob/232fc0d951e5ad096d2255381003b87f281517f7/host-bin/enter-chroot#L455) and an `exit` at line [#560](https://github.com/ibrado/crouton/blob/232fc0d951e5ad096d2255381003b87f281517f7/host-bin/enter-chroot#L560)  to see what's getting set. I posted the output at pastebin [here](https://pastebin.com/9yP7PQvV).  Hopefully, you'll get something useful out of it.

Reply:
Thanks! I guess those keys aren't set if there's only one, account set up, i.e. `jq -r ".profile.info_cache | to_entries[]" /home/chronos/Local\ State` would probably only have the default entry. Sorry I couldn't test this myself, only have one Chromebook and am hesitant to powerwash, hehe.

Anyway, I've made it so it always checks for an empty `$localdownloads`. Just committed it...

Reply:
Cool, that works. 👍

Reply:
BTW, I ran both commits on a Chromebook with multiple users and they both worked so I guess it was good to check it on a single user device.

Reply:
Woohoo! Yeah, it was good you had that around... I was seriously considering a powerwash earlier. :-P

Reply:
I have four Chromebooks going all the way back to a Cr-48 and now a Pixelbook so I have options. ,-)
I like to run the 'chrx' script, 1st phase only and use partiton 7 as a safe place for crouton so I don't have to worry about powerwashing.

Reply:
Haven't had the guts to try chrx yet, haha! I had 2 Chromebooks before this one, but I lent one and gave the other away...